src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -885,6 +885,7 @@ static TypeAccelerators()
             builtinTypeAccelerators.Add("psnoteproperty", typeof(PSNoteProperty));
             builtinTypeAccelerators.Add("psaliasproperty", typeof(PSAliasProperty));
             builtinTypeAccelerators.Add("psvariableproperty", typeof(PSVariableProperty));
+            builtinTypeAccelerators.Add("actionpreference", typeof(ActionPreference));
```

Comment:
It was discussed [around here](https://github.com/PowerShell/PowerShell/pull/8205#issuecomment-438895740). There was some resistance, a rebuttal, and then no official judgement.

Personally, having used this a lot in scripts written with this PR build, and having tried to use this a lot in scripts where it was not present, I really think it should be there by default -- it's very, very useful for error handling, which should be easy to write and easy to see in scripts. You only get both of those with the type accelerator. That said, I'll do whatever is deemed appropriate by the committee.

Reply:
FWIW, static `ActionPreference` types as strings can be typoed in any scope other than global because the variable will not automatically be strongly typed in a child scope (easy to miss bugs). Plus, they are not as visible in scripts as `[actionpreference]::ignore` is.

e.g.

```powershell
# This is possible
function Test-Something {
    [CmdletBinding()]
    param()
    $ErrorActionPreference = 'SilenltyContinue' # Whoops, typo, but PS allows it
    # ... do stuff here
}
Test-Something

# This is not with the type accelerator
function Test-Something {
    [CmdletBinding()]
    param()
    $ErrorActionPreference = [actionpreference]::SilentlyContinue
    # ... do stuff here
}
Test-Something
```

Reply:
Hmm, this probably should be addressed by making `ErrorActionPreference` an `AllScope` variable. But `AllScope` variable seems not working as expected ... need to look into that.
Anyways, I will get this addition reviewed by the committee.

Reply:
Not `AllScope`. Variables with `AllScope` are referenced in every scope, so they're singletons. Preference variables need to be modifiable in child scopes without changing the value in a parent scope, so they can't be singletons. PowerShell doesn't have a way to tell it that a global automatic variable should always be of a certain type whenever it is used in another scope, which is what would be needed to prevent the possibility of typos as shown above, but to apply logic like that to preference variables would definitely be breaking at this point.

Reply:
If I recall correctly, the original concern about adding an accelerator is perf impact of looking to see if the type matched an accelerator.  However, since accelerators are stored in a Dictionary, adding another entry in the Dictionary has a negligible impact on the perf.  So the question then is if it's valuable enough to add since if we add it, we can't remove it without it being a breaking change.  It does seem useful to aid in tab-completion and avoiding typos.  So I'm ok with adding the accelerator.

Reply:
> It does seem useful to aid in tab-completion

I agree that this is very valuable scenario.